pkg/events/event_publisher.go

```diff
@@ -68,6 +77,43 @@ func (sp *EventPublisher) Stop() {
 	sp.stop.Store(true)
 }
 
+func (sp *EventPublisher) AggregateAppTrackedResourceFromEvents(messages []*si.EventRecord) {
```

Comment:
This is very complicated for something which isn't even a problem in my opinion. 

I'm a strong -1 for optimizing tracking this way.